src/components/Map/index.jsx

```diff
@@ -13,52 +14,80 @@ class Map extends Component {
   _isMounted = false;
 
   state = {
-    map: {
-      latitude: 49.2608838,
-      longitude: -123.1139269,
-      zoom: 13
+    user: {
+      latitude: 0,
+      longitude: 0
     },
     pollingStations: [
       {
         additionalInfo: '',
         address: '',
         election: null,
-        electionId: 1,
+        electionId: 0,
         generalAccessInfo: null,
         latitude: 0,
         longitude: 0,
         name: '',
         parkingInfo: null,
         pollingStationId: 0,
         washroomInfo: null,
-        wheelchairInfo: null
+        wheelchairInfo: null,
+        distance: 0
       }
     ],
     markers: []
   };
 
   componentDidMount() {
     this._isMounted = true;
-    const { longitude, latitude, zoom } = this.state.map;
-
-    this._map = new mapboxgl.Map({
-      container: 'map',
-      style: 'mapbox://styles/mapbox/streets-v11',
-      center: [longitude, latitude],
-      zoom
-    });
-    this.liveGetCenter();
+    this.initializeMap();
     this.loadApiData().then(() => {
+      this.sortPollingStationsByDistance();
       this.renderMarkers();
     });
-
-    this.getUserLocation();
   }
 
   componentWillUnmount() {
     this._isMounted = false;
   }
 
+  initializeMap = () => {
+    this._map = new mapboxgl.Map({
+      container: 'map',
+      style: 'mapbox://styles/mapbox/streets-v11',
+      center: [0, 0],
+      zoom: 13
+    });
+    this.flyToClickedLocation();
+    this.getUserLocation();
+  };
+
+  flyToClickedLocation = () => {
+    this._map.on('click', e => {
+      this._map.flyTo({ center: e.lngLat, speed: 0.25 });
+    });
+  };
+
+  getUserLocation = () => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(position => {
+        const { latitude, longitude } = position.coords;
+        if (this._isMounted) {
+          this.setState({
+            user: {
+              latitude,
+              longitude
+            }
+          });
+
+          this._map.setCenter([longitude, latitude]);
+        }
+      });
+    } else {
+      console.warn('Geolocation is not supported by this browser.');
```

Comment:
Should there be a message on the page to warn the user? Like if the map doesn't load maybe put a message over where the map should be.

Reply:
I forgot to write it down, sorry! I wanted to do it after with better UI. Maybe like "please enable location services" or something?